pennylane/utils.py

```diff
@@ -323,6 +326,22 @@ def circuit2():
             + ",".join(string_reps)
         )
 
+    if qml.tape_mode_active():
+        for op in operation_list:
+            try:
+                qml.tape.QueuingContext.remove(op)
+            except KeyError:
+                pass
+
+        with qml.tape.QuantumTape() as tape:
+            for o in operation_list:
+                o.queue()
+                if o.inverse:
+                    o.inv()
+
+        tape.inv()
```

Comment:
Doesn't this invert the queue _and_ the gates, so that it doubles up on the above inversion?

Reply:
The step above on line 344 is only inverting operators that the user has already inverted.

E.g., if the user passes `[qml.RX, qml.RY.inv(), qml.RZ]`, then `o.inv()` is only being called for the RY gate.

This is because `.inv()` _doesn't_ change the state of the operation at all - all it does is annotate the queue. So we need to re-annotate the queue here.

It is only this line, 346, that does the full inversion.

Reply:
Thanks for explaining. I wonder if we can mark this for revision? For example use `annotate_inv` istead of the name `inv`? And make inversion something that can be toggled? Should not be hard, no?